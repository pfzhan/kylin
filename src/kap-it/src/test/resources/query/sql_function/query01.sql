--
-- Copyright (C) 2020 Kyligence Inc. All rights reserved.
--
-- http://kyligence.io
--
-- This software is the confidential and proprietary information of
-- Kyligence Inc. ("Confidential Information"). You shall not disclose
-- such Confidential Information and shall use it only in accordance
-- with the terms of the license agreement you entered into with
-- Kyligence Inc.
--
-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-- "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-- LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-- A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-- OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-- SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-- LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-- DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-- THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-- OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
--

-- test spader function
-- from KE query/sql_sparder_function/query01.sql
-- issue #6152 commit 0bb8203

SELECT CAST(MOD(ITEM_COUNT, 2) AS BIGINT) AS "mod", LN(ITEM_COUNT) AS "ln"
	, round(LOG10(ITEM_COUNT), 6) AS "log10"
	, EXP(ITEM_COUNT) AS "exp", ACOS(ITEM_COUNT) AS "acos"
	, ASIN(ITEM_COUNT) AS "asin", ATAN(ITEM_COUNT) AS "atan"
	, ATAN2(ITEM_COUNT, 0.8) AS "atan2"
	, COS(ITEM_COUNT) AS "cos", DEGREES(ITEM_COUNT) AS "degrees"
	, RADIANS(ITEM_COUNT) AS "radians", SIGN(ITEM_COUNT) AS "sign"
	, TAN(ITEM_COUNT) AS "tan", SIN(ITEM_COUNT) AS "sin"
FROM TEST_KYLIN_FACT
where ITEM_COUNT is not null and ITEM_COUNT > 10
GROUP BY ITEM_COUNT
ORDER BY "mod"
