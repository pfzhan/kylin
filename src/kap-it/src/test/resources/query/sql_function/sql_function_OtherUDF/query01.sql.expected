

SELECT "LSTG_FORMAT_NAME" FROM "TEST_KYLIN_FACT"
WHERE rlike(substr("LSTG_FORMAT_NAME",1), 'ABIN')